examples/ladder_filter.rs

```diff
@@ -0,0 +1,240 @@
+        let g3 = 1. / (1. + self.params.g.get() * a[4]);
+        //  these are just factored out of the feedback solution. Makes the math way easier to read
+        let f3 = self.params.g.get() * a[3] * g3;
+        let f2 = self.params.g.get() * a[2] * g2 * f3;
+        let f1 = self.params.g.get() * a[1] * g1 * f2;
+        let f0 = self.params.g.get() * g0 * f1;
+        // outputs a 24db filter
+        self.vout[3] = (f0 * input * a[0]
+            + f1 * g0 * self.s[0]
+            + f2 * g1 * self.s[1]
+            + f3 * g2 * self.s[2]
+            + g3 * self.s[3])
+            / (f0 * self.params.res.get() * a[3] + 1.);
+        // since we know the feedback, we can solve the remaining outputs:
+        self.vout[0] = g0
+            * (self.params.g.get()
+                * a[1]
+                * (input * a[0] - self.params.res.get() * a[3] * self.vout[3])
+                + self.s[0]);
+        self.vout[1] = g1 * (self.params.g.get() * a[2] * self.vout[0] + self.s[1]);
+        self.vout[2] = g2 * (self.params.g.get() * a[3] * self.vout[1] + self.s[2]);
+    }
+    // linear version without distortion
+    pub fn run_ladder_linear(&mut self, input: f32) {
+        // denominators of solutions of individual stages. Simplifies the math a bit
+        let g0 = 1. / (1. + self.params.g.get());
+        let g1 = self.params.g.get() * g0 * g0;
+        let g2 = self.params.g.get() * g1 * g0;
+        let g3 = self.params.g.get() * g2 * g0;
+        // outputs a 24db filter
+        self.vout[3] = (g3 * self.params.g.get() * input
+            + g0 * self.s[3]
+            + g1 * self.s[2]
+            + g2 * self.s[1]
+            + g3 * self.s[0])
+            / (g3 * self.params.g.get() * self.params.res.get() + 1.);
+        // since we know the feedback, we can solve the remaining outputs:
+        self.vout[0] =
+            g0 * (self.params.g.get() * (input - self.params.res.get() * self.vout[3]) + self.s[0]);
+        self.vout[1] = g0 * (self.params.g.get() * self.vout[0] + self.s[1]);
+        self.vout[2] = g0 * (self.params.g.get() * self.vout[1] + self.s[2]);
+    }
+}
+impl LadderParameters {
+    pub fn set_cutoff(&self, value: f32) {
+        // cutoff formula gives us a natural feeling cutoff knob that spends more time in the low frequencies
+        self.cutoff.set(20000. * (1.8f32.powf(10. * value - 10.)));
+        // bilinear transformation for g gives us a very accurate cutoff
+        self.g
+            .set((PI * self.cutoff.get() / (self.sample_rate.get())).tan());
+    }
+    pub fn get_poles(&self) -> f32 {
+        self.poles.load(Ordering::Relaxed) as f32 + 1.
+    }
+}
+impl PluginParameters for LadderParameters {
+    fn get_parameter(&self, index: i32) -> f32 {
+        match index {
+            0 => self.cutoff.get(),
+            1 => self.res.get(),
+            2 => self.get_poles(),
+            3 => self.drive.get(),
+            _ => 0.0,
+        }
+    }
```

Comment:
Can you elaborate on what you mean? It seems unnecesary now that there's get_parameter_text, so am I just supposed to delete it?

Reply:
```rust
let param_val = 0.5;
plugin.set_parameter(3, param_val);

let param_val_from_plugin = plugin.get_parameter(3);
assert_eq!(param_val_from_plugin, param_val);
```

In this case, the `assert_eq!()` would fail because you're multiplying param 3 (drive) by 5.f32 in `set_parameter()`, so you'll be returning `2.5f32` from `get_parameter()`.

Reply:
Ah I see. Didn't know they get_parameter had to return the original value from set_parameter. Don't love my solution but it's fixed now.

Reply:
Doesn't necessarily have to be the same for discrete parameters (like enums), but in general yes, round trip should be equal.

Reply:
Solution looks good to me, thanks for taking care of it :+1: